### PR TITLE
[Az.RecoveryServices.Backup] Added support for Archive smart tiering for AzureVM and MSSQL workloads

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Added support for Archive smart tiering for AzureVM and MSSQL workloads

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1218

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/RetentionPolicyConversions.cs [hunk 1]
```diff
@@ -627,6 +627,66 @@ private static ServiceClientModel.YearlyRetentionSchedule GetServiceClientLTRYea
...
+        #region Tiering Policy conversions 
+
+        /// <summary>
+        /// Helper function to convert service tiering policy from ps tiering policy.
+        /// </summary>
+        public static IDictionary<string, ServiceClientModel.TieringPolicy> GetServiceClientTieringPolicy(
+            TieringPolicy tierPolicy)
+        {
+            /*if (tierPolicy == null) // check 
+            {
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/RetentionPolicyConversions.cs:639]: REMOVE

File: src/RecoveryServices/RecoveryServices.Backup.ServiceClientAdapter/RecoveryServices.Backup.ServiceClientAdapter.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
   <ItemGroup>
 	  <PackageReference Include="Microsoft.Azure.Management.RecoveryServices" Version="4.3.3-preview" />
 	  <PackageReference Include="Microsoft.Azure.Management.RecoveryServices.Backup" Version="5.0.0-preview" />
+	  <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="3.17.4-preview" />
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices.Backup.ServiceClientAdapter/RecoveryServices.Backup.ServiceClientAdapter.csproj:17]: The static issue basically means when you introduce a new reference, you need to add the assembly of the library to the "RequiredAssemblies" filed of your module's psd1.
However we have generated some common clients so the SDK isn't necessary.
Can you use this [FeatureClient](https://github.com/Azure/azure-powershell-common/blob/main/src/ResourceManager/Version2016_09_01/Generated/FeatureClient.cs#L36) instead?

Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.ServiceClientAdapter/RecoveryServices.Backup.ServiceClientAdapter.csproj:17]: sure, let me check. Thanks!

**Review by isra-fel — APPROVED** [review]

**Comment by isra-fel** [comment]
This branch has conflicts that must be resolved @hiaga

**Review by isra-fel — APPROVED** [review]

**Pull request merged**